### ISSUE TITLE
master consideres batchnorm mean/std update as gradients

### ISSUE THREAD

**Issue by vlimant** [vlimant/mpi_learn#19]
in the way the master receives the "update" from the workers, for the bn mean weight (the running mean) it would consider the diff as a gradient and do something with this, instead of applying something dedicated to a value that was not updated by gradient descent.
the gamma/beta weights of the bn are ok in this respect.
I fear that this is playing badly with https://github.com/svalleco/mpi_learn/pull/3 
@duanders if you have any insights on how to modify the mpi-learn-optimizers to take this in consideration, please do tell

**Comment by duanders** [comment]
I see, glad you got to the bottom of it.  Have you checked how the keras optimizers handle this?

**Comment by vlimant** [comment]
I was not able to track this down all the way
